HTTP_API.md

```diff
@@ -0,0 +1,952 @@
+    `GET /api/v1/recordings/:recordingName`
+
+    `recordingName` - The name of the saved recording to get.
+
+    ###### response
+    `200` - The body is an octet stream consisting of the requested recording.
+
+    `401` - User authentication failed. The body is an error message.
+
+    `404` - The recording could not be found. The body is an error message.
+
+    `407` - JMX authentication failed. The body is an error message.
+
+    `500` - The recording was found but it could not be written to the response.
+    Or there was an unexpected error. The body is an error message.
+
+    ###### example
+    ```
+    $ curl localhost:8181/api/v1/recordings/localhost_foo_20200910T214559Z.jfr --output foo.jfr
+      % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                     Dload  Upload   Total   Spent    Left  Speed
+    100  942k  100  942k    0     0  61.3M      0 --:--:-- --:--:-- --:--:-- 61.3M
+    ```
+
+
+* #### `RecordingsGetHandler`
+
+    ###### synopsis
+    Returns a list of the recordings that are saved in persistent storage.
+
+    ###### request
+    `GET /api/v1/recordings`
+
+    ###### response
+    `200` - The body consists of a comma-seperated list of recordings,
+    enclosed in square brackets.
+
+    The format for a recording is
+    `{"downloadUrl":"$DOWNLOAD_URL","name":"$NAME","reportUrl":"$REPORT_URL"}`.
+
+    So a list with no recordings looks like `[]`,
+    a list with one recording looks like `[{...}]`,
+    and a list with two recordings looks like `[{...},{...}]`.
+
+    `401` - User authentication failed. The body is an error message.
+
+    `407` - JMX authentication failed. The body is an error message.
+
+    `500` - There was an unexpected error. The body is an error message.
+
+    `501` - The archive path where recordings are saved could not be accessed.
+    The body is an error message.
+
+    ###### example
+    ```
+    $ curl localhost:8181/api/v1/recordings
+    [{"downloadUrl":"http://192.168.0.109:8181/api/v1/recordings/localhost_foo_20200903T202547Z.jfr","name":"localhost_foo_20200903T202547Z.jfr","reportUrl":"http://192.168.0.109:8181/api/v1/reports/localhost_foo_20200903T202547Z.jfr"}]
+    ```
+
+
+* #### `RecordingsPostHandler`
+
+    ###### synopsis
+    Uploads a recording from the client to Container JFR's persistent storage.
+
+    ###### request
+    `POST /api/v1/recordings`
+
+    The recording should be uploaded in a form with the name `recording`.
+    The filename of the recording must follow the format that Container JFR
+    uses for recordings it saves itself
+    (with [`TargetRecordingPatchHandler`](#TargetRecordingPatchHandler)).
+
+    ###### response
+    `200` - The body is `{"name":"$NAME"}`, where `$NAME` is the name of the
+    recording that is now saved in persistent storage.
+    This name will be different from the uploaded recording's filename
+    if another recording with the same name already existed.
```

Comment:
Please specify how the filename is mutated when this collision-avoidance is performed